src/proper_statem.erl

```diff
@@ -462,6 +510,189 @@ move_shrinker(Seq, Par, I) ->
 	 move_shrinker(NewSeq, NewPar, I-1)).
 
 
+%% -----------------------------------------------------------------------------
+%% Targeted command generation
+%% -----------------------------------------------------------------------------
+
+
+%% @doc A special PropEr type which generates random command sequences,
+%% according to an abstract state machine specification and taking into
+%% consideration a utility value. The function takes as input the name of
+%% a callback module, which contains the state machine specification. The
+%% initial state is computed by `Mod:initial_state/0'.
+
+-spec targeted_commands(mod_name()) -> proper_types:type().
+targeted_commands(Mod) ->
+  proper_types:add_prop(is_user_nf_stateful, true,
+                        ?USERNF(targeted_commands_gen(Mod),
+                                next_commands_gen(Mod))).
+
+%% @doc Similar to {@link targeted_commands/1}, but generated command sequences
+%% always start at a given state. In this case, the first command is always
+%% `{init, InitialState}' and is used to correctly initialize the state
+%% every time the command sequence is run (i.e. during normal execution,
+%% while shrinking and when checking a counterexample). In this case,
+%% `Mod:initial_state/0' is never called.
+
+-spec targeted_commands(mod_name(), symbolic_state()) -> proper_types:type().
+targeted_commands(Mod, InitialState) ->
+  proper_types:add_prop(is_user_nf_stateful, true,
+                        ?USERNF(targeted_commands_gen(Mod, InitialState),
+                                next_commands_gen(Mod, InitialState))).
+
+%% @private
+-spec targeted_commands_gen(mod_name()) -> proper_types:type().
+targeted_commands_gen(Mod) ->
+  ?LET(Cmds,
```

Comment:
In this and the following three functions decide whether you want `?LET` to span three or just two lines. (two is OK IMO).